src/utils/Processing.ts

```diff
@@ -44,17 +44,16 @@ export function findInputDir(input: string): string {
   const originalInput = input;
 
   const inputFshSubdirectoryPath = path.join(originalInput, 'input', 'fsh');
-  const fshFilesPresent = checkForFshFiles(originalInput);
   const fshSubdirectoryPath = path.join(originalInput, 'fsh');
   const rootIgDataPath = path.join(originalInput, 'ig-data');
   const currentTankWithNoFsh =
-    !fshFilesPresent && !fs.existsSync(fshSubdirectoryPath) && !fs.existsSync(rootIgDataPath);
+    !fs.existsSync(fshSubdirectoryPath) &&
+    !fs.existsSync(rootIgDataPath) &&
+    !hasFshFiles(originalInput);
 
-  if (fs.existsSync(inputFshSubdirectoryPath)) {
-    // Use input/fsh/ subdirectory if not already specified and present
-    input = path.join(originalInput, 'input', 'fsh');
-  } else if (currentTankWithNoFsh) {
-    // Use input/fsh/ subdirectory when in the current tank configuration without FSH files
+  // Use input/fsh/ subdirectory if not already specified and present
+  // or when in the current tank configuration without FSH files
+  if (fs.existsSync(inputFshSubdirectoryPath) || currentTankWithNoFsh) {
```

Comment:
I hate to be annoying, but I'm realizing now that `currentTankWithNoFsh` does not check `inputFshSubdirectoryPath`.  So in our most common use case (a project w/ `input/fsh`), the assignment to `currentTankWithNoFsh` will still cause `hasFshFiles(...)` to be invoked even though `currentTankWithNoFsh` isn't even looked at if `input/fsh` exists.

So, to really avoid that expensive call, you'd need to either include `!fs.existsSync(inputFshSubdirectoryPath)` in the assignment of `currentTankWithNoFsh` or you'd need to make an `isCurrentTankWithNoFsh` _function_ and call that on line 56 above (so it isn't called if `input/fsh` exists).

Reply:
Sorry I missed that case when I reviewed the first time.  I actually had thought of it, but then for some reason thought it was fine after reviewing the code.  I think maybe I saw `fshSubdirectoryPath` and got it mixed up with `inputFshSubdirectoryPath` (or something... who knows).

Reply:
No worries I should have thought of that case as well. I went with the first approach you described and added the check for `!fs.existsSync(inputFshSubdirectoryPath)` to the assignment of `currentTankWithNoFsh` because it seemed to make more sense to me mentally, but I'm happy to update it if you prefer the other approach. Fixed in 4d8f692.

Also, if the variable names are getting confusing, I can try renaming them.